core/src/backend/audio.rs

```diff
@@ -532,6 +532,38 @@ impl<'gc> AudioManager<'gc> {
         });
     }
 
+    /// Stops any sound associated with the given Display Object and its children. The DO must represent the parent.
+    /// Sounds associated with DOs are an AVM1/Timeline concept and should not be called from AVM2 scripts.
+    pub fn stop_sounds_on_parent_and_children(
+        &mut self,
+        audio: &mut dyn AudioBackend,
+        display_object: DisplayObject<'gc>,
+    ) {
+        self.sounds.retain(move |sound| {
+            if let Some(other) = sound.display_object {
+                let stop = if DisplayObject::ptr_eq(other, display_object) {
```

Comment:
Could this be merged into something like
```
let mut other_do = Some(other);
while let Some(other) = other_do {
    if DisplayObject::ptr_eq(other_do, display_object) {
        audio.stop_sound(sound.instance);
        return false;
    }
    other_do = other.parent();
}
```
?

Reply:
Nice solution! It's shorter and cleaner, and I believe it should work. I'll give it a try tomorrow. 😉